pkg/proxy/util/utils.go

```diff
@@ -123,23 +123,36 @@ func IsProxyableHostname(ctx context.Context, resolv Resolver, hostname string)
 	return nil
 }
 
-// IsLocalIP checks if a given IP address is bound to an interface
-// on the local system
-func IsLocalIP(ip string) (bool, error) {
+// GetLocalAddrs returns a list of all network addresses on the local system
+func GetLocalAddrs() ([]net.IP, error) {
```

Comment:
Why not return a "set" type here.  k8s.io/utils/net/ipnet.go defines `IPNetSet`.  We could have the equivalent `IPSet` type

Reply:
Makes sense to me, I'll open a PR to add `IPSet` type to k8s.io/utils/net.

Reply:
On second thought, I'm realizing that a set type for net.IP may not work because we can't do comparison checks for net.IP based on their string values since an IPv6 representation of IPv4 addresses are considered equal (see [net.IP Equals](https://golang.org/src/net/ip.go?s=10028:10057#L397)). We could just ignore this case and assume equality/uniqueness soley based on string values but that would be a behavioral change from what we have right now.

Reply:
But `String` has the same logic. Any two `net.IP`s that compare as `Equal` will also `String`-ify the same. (Specifically, `net.ParseIP("::ffff:1.2.3.4").String() → "1.2.3.4"`)

So as long as you know the IPs are in canonical form, you can compare based on string values.

Reply:
Ah good to know! I'll update k8s.io/utils/net to add a set type for net.IP.